### ISSUE TITLE
Init State of TabBarView Widgets called more then once despite using AutomaticKeepAliveMixin

### ISSUE THREAD

**Issue by akshg05** [flutter/flutter#68961]
## Steps to Reproduce

A simple App with tabs where there are 5 TabViews as Widgets A, B, C, D, E.
Each TabView widget is a Stateful widget and uses an AutomaticKeepAliveCLientMixin to prevent the initState being called more than once(twice specifically).
When we navigate from TabView A to C, states of tabs in between are built twice, that is Widget B has initState called twice.
And still the Widget B has not loaded completely and if you revisit Widget B , init State is called again.

**Expected results:** 

I expect the initState of each widget as my TabView to be called only once since I'm using AutomaticKeepAliveClientMixin even when I jump from one tab to a distant tab.

**Actual results:** 

InitState of widgets as my TabView is called twice if jumped to a distant widget.
InitState is not called again if you have rested/landed on that tab completely.
ie. 

If we jump from Tab A -> Tab C, initState for Tab B is called twice.
Now if I visit Tab B again initState is called again (ie. making it 3 times!)
From here on, initState of B is not called any more.

<details>
  <summary>Logs</summary>

```
[✓] Flutter (Channel stable, 1.22.2, on Linux, locale en_GB.UTF-8)
    • Flutter version 1.22.2 
    • Framework revision 84f3d28555 (9 days ago), 2020-10-15 16:26:19 -0700
    • Engine revision b8752bbfff
    • Dart version 2.10.2

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /home/user/Android/Sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: /home/user/Applications/android-studio-ide-191.6010548-linux/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.
```

</details>

**Comment by pedromassangocode** [comment]
Hi @akshg05 
Can you please provide your `flutter doctor -v` and a minimal complete reproducible code sample?
Thank you

**Comment by pedromassangocode** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact description of what happens, logs, and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.
